src/api/libcellml/types.h

```diff
@@ -58,6 +58,8 @@ class Parser; /**< Forward declaration of Parser class. */
 using ParserPtr = std::shared_ptr<Parser>; /**< Type definition for shared parser pointer. */
 class Printer; /**< Forward declaration of Printer class. */
 using PrinterPtr = std::shared_ptr<Printer>; /**< Type definition for shared printer pointer. */
+class Strict; /**< Forward declaration of Parser class. */
+using StrictPtr = std::shared_ptr<Strict>; /**< Type definition for shared parser pointer. */
```

Comment:
I am not going to duplicate code that should be in a class, no matter how limited the functionality.

Reply:
I can't argue with the fact that duplicated is and should always be a no-go.

However, as it stands, the `Strict` class is part of the API while it has nothing to do there. So, rather than have the `Parser` and `Importer` classes inherit from `Strict`, I would rather have `ParserImpl` and `ImporterImpl` inherit from it, and thus not have the `Strict` class part of the API (and mentioned in `src/api/libcellml/types.h`), which is just a nonsense to me.